### PR TITLE
I795 description

### PR DESCRIPTION
**Pull request by AntonioRecaldeRusso**
description now updates correct class docstring

### CODE REVIEW

File: src/main/python/scripts/smvserver.py [hunk 1]
```diff
@@ -488,6 +499,30 @@ def buildRequiresDS(requiresDS, indentation="\t"):
 {1}def requiresDS(self):\n\
 {1}{1}return [{0}]\n".format(requires, indentation)
 
+def getRequiredModulesWithLinks(currentFqn, fqns):
```
Comment (AliTajeldin) [inline src/main/python/scripts/smvserver.py]: rename `fqns` to `requiredFqns`.

File: src/main/python/scripts/smvserver.py [hunk 2]
```diff
@@ -488,6 +499,30 @@ def buildRequiresDS(requiresDS, indentation="\t"):
...
+    '''takes a fqn (currentFqn) and a list of fqns or SmvModuleLink(s) required by the currentFqn
```
Comment (AliTajeldin) [inline src/main/python/scripts/smvserver.py]: This comment is a bit confusing.  Are we taking a list that has `SmvModuleLink(s)` or are we creating a list that has `SmvModuleLinks`?

File: src/main/python/scripts/smvserver.py [hunk 3]
```diff
@@ -488,6 +499,30 @@ def buildRequiresDS(requiresDS, indentation="\t"):
...
+    for fqn in fqns:
+        # if same stage, fqn
+        if currentStage == getStageFromFqn(fqn):
+            requiresDSList.append(fqn)
+        # if different stage, wrap around link
+        else:
+            requiresDSList.append(RequiredDSLink(fqn).varName)
+    return requiresDSList
+
+def getRequiredDSLinks(currentFqn, fqns):
```
Comment (AliTajeldin) [inline src/main/python/scripts/smvserver.py]: what is the difference between this function and the one above?
Needs a method comment to explain.

**Comment by AliTajeldin** [comment]
Btw, how did a bunch of commit for ticket #770 get added to this PR?

File: src/main/python/scripts/smvserver.py [hunk 4]
```diff
@@ -488,6 +499,30 @@ def buildRequiresDS(requiresDS, indentation="\t"):
...
+    requiresDSList = []
+    currentStage = getStageFromFqn(currentFqn)
+
+    for fqn in fqns:
+        # if same stage, fqn
+        if currentStage == getStageFromFqn(fqn):
+            requiresDSList.append(fqn)
+        # if different stage, wrap around link
+        else:
+            requiresDSList.append(RequiredDSLink(fqn).varName)
```
Comment (AliTajeldin) [inline src/main/python/scripts/smvserver.py]: Are we just creating the `RequiredDSLink` object to throw it away? seems weird.

Comment (AntonioRecaldeRusso) [inline src/main/python/scripts/smvserver.py]: It is done so to transform the fqn into the SmvModuleLink variable name. Namely, it avoids having to repeat the link `Namely "{}_link".format(fqn.replace(".", "_"))`
It is also possible to create a static method in the RequiredDSLink class that will transform the fqn into the var name, and thus avoid creating an object and throwing it away.

File: src/main/python/scripts/smvserver.py (hunk 3 above)
Comment (AntonioRecaldeRusso) [inline src/main/python/scripts/smvserver.py]: comments added e481809

**Pull request merged**